testcases/OpTestXive.py

```diff
@@ -0,0 +1,210 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+'''
+OpTestXive
+------------
+
+This module contain testcases related to XIVE.
+
+1.check for there are no xive related
+errors in dmesg log
+2.Enable kernel logging, basically generate traffic
+3.Enable console traffic by printing the 
+processes/tasks to the console
+4.check Xive is configured properly or not
+5.check system rebbot with xive=on
+6.check system reboot with xive=off
+7.validate interrupts handled by CPU's
+'''
+import time
+import subprocess
+import commands
+import re
+import sys
+import pexpect
+import os
+import time
+
+
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.OpTestError import OpTestError
+from common import OpTestHMC
+
+import unittest
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+from common.OpTestSSH import ConsoleState as SSHConnectionState
+
+class OpTestXive(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.platform = conf.platform()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        self.cv_HMC = self.cv_SYSTEM.hmc
+
+    def xive_test(self):
+        '''
+        This function will check for 
+        1.Make sure no errors in dmesg log
+        2.Enable kernel logging, basically generate traffic
+        3.Enable console traffic by printing the 
+        processes/tasks to the console
+        '''
+    	self.console = self.cv_HMC.get_console()
+        self.cv_HMC.vterm_run_command(self.console, "dmesg -T --level=alert,crit,err,warn | grep -i xive")
+        self.cv_HMC.vterm_run_command(self.console, "dmesg -T --level=alert,crit,err,warn | grep -i irq")
+        self.cv_HMC.vterm_run_command(self.console, "echo 10 > /proc/sys/kernel/printk")
+        i = 1
+        while i <= 20: 
+        	self.cv_HMC.vterm_run_command(self.console, "echo t > /proc/sysrq-trigger")
+		i = i+1
+        self.cv_HMC.vterm_run_command(self.console, "echo $?")
+
+class XiveTesting(OpTestXive):
+    '''
+    1.This function will check xive is configured properly or not
+    2.check for system reboot with Xive=off
+    3.check for system reboot with Xive=on
+    '''
+    def setup_test(self):
+        self.console = self.cv_HMC.get_console()
+        self.cv_HMC.vterm_run_command(self.console, "uname -a")
+        result = self.cv_HMC.vterm_run_command(self.console, "grep -i --color=never xive /boot/config-`uname -r`")
+        print result
+        for line in result:
+           temp = ''.join(self.cv_HMC.vterm_run_command(self.console, "echo %s | cut -d '=' -f 2" % line))
+           if(temp != 'y'):
+               self.fail("xive is not configured properly")
+               '''
```

Comment:
why is these all lines are kept under docstring, mistakenly?

Reply:
yes by mistake